apps/src/tests/Test3282.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect } from 'react';
+import { View, Text, Button, StyleSheet } from 'react-native';
+import { NavigationContainer } from '@react-navigation/native';
+import { createNativeStackNavigator, NativeStackNavigationProp } from '@react-navigation/native-stack';
+import { featureFlags } from 'react-native-screens';
+
+const SYNCHRONOUS_STATE_UPDATES_ENABLED = true;
+
+type StackParamList = {
+  Home: undefined,
+  FormSheet: undefined,
+}
+
+type RouteProps = {
+  navigation: NativeStackNavigationProp<StackParamList>;
+}
+
+const Stack = createNativeStackNavigator();
+
+const HomeScreen = ({ navigation }: RouteProps) => {
+  useEffect(() => {
+    featureFlags.experiment.synchronousScreenUpdatesEnabled = SYNCHRONOUS_STATE_UPDATES_ENABLED
+    featureFlags.experiment.synchronousHeaderConfigUpdatesEnabled = SYNCHRONOUS_STATE_UPDATES_ENABLED
+    featureFlags.experiment.synchronousHeaderSubviewUpdatesEnabled = SYNCHRONOUS_STATE_UPDATES_ENABLED
+  }, [])
```

Comment:
I haven't notice earlier, but we don't want to modify these during render. This is not a good example. These should be enabled / disabled in top level scope.

Reply:
https://github.com/software-mansion/react-native-screens/pull/3282/commits/1c09c040712f8aec28906b674a7790814841b1d9